syntax = "proto3";

import "map_renderer.proto";
import "transport_router.proto";

package transport_catalogue;

message Stop {
    uint32 id = 1;
    string stop_name = 2;
    double lat = 3;
    double lng = 4;
}

message Bus {
    string bus_name = 1;
    repeated uint32 stops = 2;
    bool is_roundtrit = 3;
}

message StopToStopDist {
    uint32 stop_from = 1;
    uint32 stop_to = 2;
    uint32 distance = 3;
}

message TransporCatalogue {
    repeated Stop stops = 1;
    repeated Bus buses = 2;
    repeated StopToStopDist from_stop_to_stop = 3;
}

message ComplexCatalogueInfo {
    TransporCatalogue catalogue = 1;
    transport_map_render.RenderSettings map_render_settings = 2;
    transport_router.Router router = 3;
}